mesonbuild/compilers/c.py

```diff
@@ -864,7 +860,19 @@ def find_library_impl(self, libname, env, extra_dirs, code, libtype='default'):
                     trial = os.path.join(d, prefix + libname + '.' + suffix)
                     if os.path.isfile(trial):
                         return [trial]
-        return None
+        # XXX: For OpenBSD and macOS we (may) need to search for libfoo.x.y.z.dylib
+        return []
+
+    def find_library_impl(self, libname, env, extra_dirs, code, libtype):
+        # These libraries are either built-in or invalid
+        if libname in self.ignore_libs:
+            return []
+        if isinstance(extra_dirs, str):
+            extra_dirs = [extra_dirs]
+        key = (tuple(self.exelist), libname, tuple(extra_dirs), code, libtype)
+        if not key in self.find_library_cache:
```

Comment:
__[Flake8]__

[E713] test for membership should be 'not in'


<!-- Comment by [SideCI](https://sideci.com) -->